x-pack/legacy/plugins/uptime/public/state/api/ml_anomaly.ts

```diff
@@ -18,7 +18,23 @@ import {
 import { DataRecognizerConfigResponse } from '../../../../../../plugins/ml/common/types/modules';
 import { JobExistResult } from '../../../../../../plugins/ml/common/types/data_recognizer';
 
-export const getMLJobId = (monitorId: string) => `${monitorId}_${ML_JOB_ID}`.toLowerCase();
+const getJobPrefix = (monitorId: string) => {
+  // ML App doesn't support upper case characters in job name
+  // Also Spaces and the characters / ? , " < > | * are not allowed
+  // so we will replace all special chars with _
+
+  const lowerCaseMonitorId = monitorId.replace(/[^A-Z0-9]+/gi, '_').toLowerCase();
+
+  // ML Job ID can't be greater than 64 length, so will be substring it, and hope
+  // At such big length, there is minimum chance of having duplicate monitor id
+  // Subtracting ML_JOB_ID constant and _ char as well
+  if ((lowerCaseMonitorId + ML_JOB_ID + 1).length > 64) {
```

Comment:
This is somewhat awkward in that in accounts for the extra `_` we add. It feels like this function should just add the `_` to the end itself. Is there a reason not to?

Reply:
yes, it's an awkward code, i have further tried simplifing it.